tests/all/vec.rs

```diff
@@ -167,3 +167,16 @@ fn test_vec_write() {
 
     assert_eq!(v, &[1, 2, 3]);
 }
+
+#[cfg(feature = "serde")]
+#[test]
+fn test_vec_serializes() {
+    let bump = Bump::new();
+    let mut vec = Vec::new_in(&bump);
+    vec.push("hello");
+    vec.push("world");
+    assert_eq!(
+        serde_json::to_string(&vec).unwrap(),
+        "[\"hello\",\"world\"]"
+    );
+}
```

Comment:
Similarly, can we sanity test more cases than just `Vec<&'static str>`? And can we compare against `std::vec::Vec<T>`'s serialization rather than hard coding the JSON?

Also worth deserializing the JSON into a `std::vec::Vec<T>` and asserting that the values round tripped properly. (No use in adding serialization if nothing can deserialize it.) Same for the box case.

Reply:
Okay I've added more test cases and tested them against serialize/deserialize with `std::vec::Vec`